firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatInfo.java

```diff
@@ -51,10 +86,69 @@ private DefaultHeartBeatInfo(Context context) {
     return HeartBeat.NONE;
   }
 
+  @Override
+  public Task<List<HeartBeatResult>> getAndClearStoredHeartBeatInfo() {
+    TaskCompletionSource<List<HeartBeatResult>> taskCompletionSource = new TaskCompletionSource<>();
+    this.backgroundExecutor.execute(
```

Comment:
Even though `Tasks.call()` is deprecated I suggest you use it as it makes sure that the task actually completes in the face of exceptions and errors, i.e. see b/157690112 for how tasks can hang indefinitely if an exception happens.